functional-tests/mining_chain_test.go

```diff
@@ -24,21 +22,17 @@ import (
 )
 
 func TestSingleMiner(t *testing.T) {
-	tf.FunctionalTest(t)
+	tf.IntegrationTest(t)
```

Comment:
This is good. We should should reserve the functional test flag for tests that take over 5 seconds or so. But it no longer makes sense to keep it in this file in the functional tests director. It should probably live in `miner_integration_test.go` and be called `TestMinerMineOnce`.

Reply:
I agree that we should probably re-arrange the tests a bit, but leaving that for later.